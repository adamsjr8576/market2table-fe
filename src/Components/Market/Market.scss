.market-article {
  width: 100%;
  border-bottom: 2px solid black;
  cursor: pointer;
}

.market-article-title {
  text-align: center;
}

.market-article:hover {
  background-color: lightgrey;
}
