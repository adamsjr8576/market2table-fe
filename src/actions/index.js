export const addZipCode = zipCode => ({
  type: 'ADD_ZIP_CODE',
  zipCode
});

export const addMarkets = markets => ({
  type: 'ADD_MARKETS',
  markets
});

export const addSelectedMarket = id => ({
  type: 'ADD_SELECTED_MARKET',
  id
});

export const addVendors = vendors => ({
  type: 'ADD_VENDORS',
  vendors
});

export const addSelectedVendor = id => ({
  type: 'ADD_SELECTED_VENDOR',
  id
});
