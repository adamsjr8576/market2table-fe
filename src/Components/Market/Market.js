import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import './Market.scss';
import { addSelectedMarket } from '../../actions';

export const Market = ({ id, name, address, products, schedule }) => {
  const dispatch = useDispatch();

  return (
    <article onClick={() => dispatch(addSelectedMarket(id))} className='market-article'>
      <h1 className='market-article-title'>{name}</h1>
    </article>
  )
}
