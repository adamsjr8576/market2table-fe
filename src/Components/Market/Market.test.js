import React from 'react';
import { shallow } from 'enzyme';
import { Market } from './Market';
import { addSelectedMarket } from '../../actions/index';
import { useSelector, useDispatch } from 'react-redux';

jest.mock("react-redux", () => ({
  useDispatch: () => jest.fn()
}));

describe('Market', () => {
  let dispatch = useDispatch();
  let mockAddSelectedMarket = jest.fn();
  let mockId = '321';
  let mockName = 'Frog Leap Farm';
  let wrapper;

  it('should match the snapshot with the id and name passed in', () => {
    wrapper = shallow(<Market
        id={mockId}
        name={mockName}
        addSelectedMarket={mockAddSelectedMarket}
        />);

      expect(wrapper).toMatchSnapshot()
  });

  it.skip('should invoke addSelectedMarket with an id on click', () => {
    wrapper = shallow(<Market
        id={mockId}
        name={mockName}
        addSelectedMarket={mockAddSelectedMarket}
        />);
    wrapper.find('.market-article').simulate('click');
    expect(dispatch).toHaveBeenCalled();
  });

})
