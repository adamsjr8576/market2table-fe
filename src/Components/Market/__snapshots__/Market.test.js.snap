// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Market should invoke addSelectedMarket with an id on click 1`] = `
<Link
  className="market-selected-link"
  to="/markets/321"
>
  <article
    className="market-article"
    onClick={[Function]}
  >
    <h1
      className="market-article-title"
    >
      Frog Leap Farm
    </h1>
  </article>
</Link>
`;

exports[`Market should match the snapshot with the id and name passed in 1`] = `
<Link
  className="market-selected-link"
  to="/markets/321"
>
  <article
    className="market-article"
    onClick={[Function]}
  >
    <h1
      className="market-article-title"
    >
      Frog Leap Farm
    </h1>
  </article>
</Link>
`;
